.github/workflows/tests.yml

```diff
@@ -28,26 +28,29 @@ jobs:
         run: tox
         env:
           PLATFORM: ${{ matrix.platform }}
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
         with:
-          name: coverage-results
+          name: coverage-results-${{ matrix.python-version }}
```

Comment:
The coverage results are supposed to be a single combined coverage report. I haven't actually checked if the outputs are being gathered correctly across all branches on github, but it is in tox.